R/pkg/R/serialize.R

```diff
@@ -37,24 +37,38 @@ writeObject <- function(con, object, writeType = TRUE) {
   # passing in vectors as arrays and instead require arrays to be passed
   # as lists.
   type <- class(object)[[1]]  # class of POSIXlt is c("POSIXlt", "POSIXt")
-  if (writeType) {
-    writeType(con, type)
+  isNA <- FALSE
+  if (type %in% c("integer", "character", "logical", "double", "numeric")) {
```

Comment:
otherwise it will throw something like "During startup - Warning message:
In is.na(object) :
  is.na() applied to non-(list or vector) of type 'environment'".
It have to add the limit of type of object, but there seems no other better way.